### DISCUSSION TITLE
Errors from input.http streams

### DISCUSSION THREAD

**Discussion by maardsma** [savonet/liquidsoap#1885] [Q&A]
My script accesses two mono mp3 streams (of air traffic control channels) and mixes them to one stereo stream, then outputs it to my icecast server.

The two mono streams can be slow to respond, busy servers I guess. Perhaps for this reason (timeouts) I often get "Invalid stream type!" and "Feeding stopped" messages when starting up this script. (seems like if it gets started successfully it's usually stable). Then it loops a lot switching between fallback (blank audio) and those errors when attempting to switch to the merged stream.

Questions:

1) How do I increase the timeout on input.http to see if that helps?

2) Is there a way I can tell the script to re-attempt the HTTP connection if it fails / dies / times out? Or does it do this automatically already when it's looping?

3) Is there a way I can slow down the looping of the script so it doesn't re-check the input feeds so often (stays on silent fallback a little longer)? Dead air during issues is ok as the feed is for internal use.

Thanks!

Mark

**Comment by maardsma** [comment]
I found the timeout parameter on input.http, I apologize that I missed that before.

I also noticed the errors about incorrect stream type (sometimes) happen after metadata is received. So the http connection was made before the error.

What would cause incorrect stream type on some startups but not others, hitting the same input streams?

Mark

**Comment by maardsma** [comment]
On this I added poll_delay=10.0, timeout=60.0, new_track_on_metadata=false to the input.http call and I haven't seen any more errors. I don't know which one made a difference, or why I got the Incorrect steam format errors without these parameters.

**Reply by toots** [reply]
Hi! Glad to hear that you found a solution to your problem!
